lib/github.py

```diff
@@ -155,6 +169,10 @@ def is_up_to_date():
     Returns True if the player is up to date.
     """
 
+    if not is_reachable('1.1.1.1'):
```

Comment:
I don't know about this. It just feels like an overly complicated workaround (which is also less reliable). You can't assume the device can't talk to GitHub or docker hub by checking if it can talk to cloudflare dns server. 

Just use a timeout argument on the request call.

Reply:
Even with adding a timeout argument in a request.get call, there's still some delay/latency. I'll probably try to disable retries.

Reply:
Try a `request.head()` as a check with a 1s timeout. A HEAD request shouldn't take more than a second or so.

Reply:
Tried that as well, but page loads are still slow.